### ISSUE TITLE
Stripe SCA Payments fail in the backoffice

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#5212]
## Description

New orders placed under /admin/orders/new fail, and display an error using Stripe as a payment method (pics below).

## Expected Behavior

Stripe payments should work, also in the backoffice.

## Actual Behaviour

Error messages are displayed with Stripe-SCA and Stripe-Connect (pics below)

## Steps to Reproduce

1. Login as admin to a shop or hub and go to /admin/orders/new
2. Create a new order for a given customer. Make sure this customer/user allows charges on a credit card or add one, under the user's account, /account -> Credit Cards. Place a Check under Authorised Shops, for the respective shop.
3. Fill the necessary details, and proceed to Payment Methods
4. Select a Stripe payment method and add the respective credit card data, allowed by the user.
5. Press Update -> The error messages (below) should be displayed.

## Animated Gif/Screenshot

with Stripe-SCA:
![image.png](https://images.zenhubusercontent.com/5d761b049dde650001eaf32f/cbc3e937-7702-49db-9fe3-a431aeb7e31e)

with Stripe-Connect:
![image.png](https://images.zenhubusercontent.com/5d761b049dde650001eaf32f/d1f641f0-dbe9-40a6-a94a-c450b06f20d2)

## Workaround

- Using other payment methods.
- The user places the order himself in the shopfront, where Stripe works.

## Severity
s2: a non-critical feature is broken, no workaround

## Your Environment

* This was fist observed on FR-staging, with the release v2.9.2
* Browser name and version: Firefox 75
* Operating System and version (desktop or mobile): Ubuntu 18.04

## Possible Fix

Perhaps worth checking this on another server other than FR-staging, with the release v2.9.2.

**Comment by Matt-Yorkley** [comment]
We need to verify if this is actually broken or not...

**Comment by Matt-Yorkley** [comment]
Ok, apparently this is confirmed broken in `v2.9.2`, but it looks like we haven't been manually testing this case? 

I can't see anywhere that Stripe or admin orders has been touched in this release, so it might be useful to validate whether or not this is present in `v2.9.1`?

This isn't covered by automated tests either, and apparently doesn't trigger an alert in Bugsnag. I think we need to prioritise #4639 soon

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley, 

I investigated this a bit further and staged this latest release v2.9.2, as well as v2.9.0 and v2.8.2  in Katuma. The issues above exist in previous releases and therefore are not caused by this last one, so good news.

Further observations:
- Stripe-Connect: the second pic (above) shows the absence of the input fields, for the card's data, which does not really allow to manually test if this payment method works. This is a display issue. Changing Payment Methods configurations, namely the order in which they appear, allows inserting the card data and yields a successful Stripe payments.
- Stripe SCA: payments fail in the backoffice - as already confirmed in production.

So basically, Stripe SCA payments don't work in the backoffice but Stripe-Connect payments do. For that reason I've changed the title of the bug - I am pinging @lin-d-hop and @RachL who were following the the discussion on Slack.

I wonder if PR #4786 also has a positive impact on this issue @luisramos0. I feel I should prioritize this test and find out.

And you are right Matt, #4639 is of importance. Also relevant, as we once discussed, to automatize different card types #5014. I hope to be able to progress on this soon.

**Comment by luisramos0** [comment]
great testing Filipe!
For now, StripeSCA is beta, so we should separate the issues with StripeSCa and StripeConnect clearly. 
No StripeSCA issue will be an S2 for now.
This was marked as S2 so lets consider the scope of this issue to be StripeConnect ok?

I think it can still be an S2, right? I am not sure I understand what do you mean by "Changing Payment Methods configurations, namely the order in which they appear". Can you clarify?

**Comment by luisramos0** [comment]
Maybe we should create a separate issue for StripeSCA.

**Comment by luisramos0** [comment]
This is broken since #4757
I havent flagged that BO payments had to be tested there because I forgot about it.

We will need a small patch to adapt StripeSCA BO payments. Basically, the payments are done in two steps: authorize and confirm. The BO code is still not aware of that for StripeSCA payments.

Another case we need to test is when we cancel a StripeSCA payment in the Backoffice.

**Comment by filipefurtad0** [comment]
I didn't think of testing this in the BO either - this was @lin-d-hop's great release-testing catch :muscle:  
Adding "cancelling stripe SCA payments in the BO" to our test suite, thanks for pointing this out @luisramos0

**Comment by lbwright22** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Test paying by Stripe in Backend - does the postcode need to be 19 chars? DO we have an issue for this?](https://trello.com/c/Jv70uBtR/900-test-paying-by-stripe-in-backend-does-the-postcode-need-to-be-19-chars-do-we-have-an-issue-for-this)

**Comment by luisramos0** [comment]
@lbwright22 this issue is about Stripe SCA the new stripe payment method that I believe you are not using yet.
